CRM/Contribute/Form/Contribution/Main.php

```diff
@@ -279,9 +279,12 @@ public function setDefaultValues() {
       }
 
       //set custom field defaults
+      $customFieldType = ['Membership'];
       foreach ($this->_fields as $name => $field) {
         if ($customFieldID = CRM_Core_BAO_CustomField::getKeyID($name)) {
```

Comment:
```suggestion
        if ($customFieldID = CRM_Core_BAO_CustomField::getKeyID($name)) {
        // check if the custom field is on a membership, we only want to load defaults for membership custom fields here, not contact fields
```

Reply:
Maybe worth a comment here, since this is not at all readable and it's pretty weird